packages/automerge-repo-storage-nodefs/src/index.ts

```diff
@@ -4,34 +4,62 @@ import { StorageAdapter } from "@automerge/automerge-repo"
 export class NodeFSStorageAdapter implements StorageAdapter {
   directory: string
 
-  constructor(directory = ".amrg") {
+  constructor(directory = "automerge-repo") {
     this.directory = directory
   }
+  loadRange(keyPrefix: string[]): Promise<Uint8Array[]> {
+    const keyString = keyPrefix.join(".")
+    return new Promise<Uint8Array[]>(resolve => {
+      fs.readdir(this.directory, (err, files) => {
+        files
+          .filter(file => file.startsWith(keyString))
+          .map(file => {
+            fs.readFile(`${this.directory}/${file}.amrg`, (err, data) => {
+              if (err) throw err
+              resolve([data])
+            })
+          })
+      })
+    })
+  }
+  removeRange(keyPrefix: string[]): void {
+    const keyString = keyPrefix.join(".")
+    fs.readdir(this.directory, (err, files) => {
+      files
+        .filter(file => file.startsWith(keyString))
+        .map(file => {
+          fs.rm(`${this.directory}/${file}.amrg`, err => {
+            if (err) throw err
+          })
+        })
+    })
+  }
 
-  fileName(docId: string) {
-    return `${this.directory}/${docId}.amrg`
+  fileName(key: string[]) {
+    const keyString = key.join(".")
+    return `${this.directory}/${keyString}.amrg`
   }
 
-  load(docId: string): Promise<Uint8Array | null> {
+  load(key: string[]): Promise<Uint8Array | undefined> {
     return new Promise<Uint8Array | null>(resolve => {
-      fs.readFile(this.fileName(docId), (err, data) => {
+      fs.readFile(this.fileName(key), (err, data) => {
         if (err) resolve(null)
         else resolve(data)
       })
     })
   }
 
-  save(docId: string, binary: Uint8Array): void {
-    fs.writeFile(this.fileName(docId), binary, err => {
+  save(key: string[], binary: Uint8Array): void {
+    fs.writeFile(this.fileName(key), binary, err => {
       // TODO: race condition if a load happens before the save is complete.
```

Comment:
hack to fix race cond:

this.isSavingPromise = Promise.resolve()

when saving, wrap isSavingPromise in a new promise that resolves once saving is complete.

When loading, await this.isSavingPromise first.

Reply:
i rewrote this extensively to make the problem go away